File: algorithm/cpp_playground/1_6.go
package main

import "fmt"

// 大きい方の数値を返す
func max(x, y int) int {
	if x > y {
		return x
	} else {
		return y
	}
}

func solve() {
	n := 5
	a := []int{2, 3, 4, 5, 10}

	ans := 0

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			for k := j + 1; k < n; k++ {
				len := a[i] + a[j] + a[k]
				ma := max(a[i], max(a[j], a[k]))
				rest := len - ma

				if ma < rest {
					ans = max(ans, len)
				}
			}
		}
	}

	fmt.Printf("%d\n", ans)
}

func main() {
	solve()
}

File: algorithm/cpp_playground/1_6.py
def solve():
    n = 5
    a = [2, 3, 4, 5, 10]

    ans = 0

    for i in range(n):
        for j in range(i + 1, n):
            for k in range(j + 1, n):
                # 辺の長さ
                len = a[i] + a[j] + a[k]
                # 一番長い辺の長さ
                ma = max(a[i], max(a[j], a[k]))
                # 他の辺の長さの和
                rest = len - ma

                # 三角形が作れる条件を満たしている
                if ma < rest:
                    # より長ければ更新
                    ans = max(ans, len)

    print(ans)


if __name__ == "__main__":
    solve()
